Bon déception, j'ai un `AttributeError: 'NoneType' object has no attribute 'items'` dont je n'arrive pas a me sortir en implémentant ton code...

Je suis preneur d’idée (on dirait que le `self`/`super` ne suis pas et donc que les membres du parent ne veulent pas venir)...

```
Traceback (most recent call last):
  File "/home/ubuntu/workspace/zds-site/zds/utils/tests/test_highlighter.py", line 16, in test_valid_highlighter
    hl.highlight(self.my_text)
  File "/home/ubuntu/workspace/zds-site/zds/utils/highlighter.py", line 11, in highlight
    super(SearchHighlighter, self).highlight(*args, **kwargs)
  File "/home/ubuntu/workspace/zds-site/zdsenv/local/lib/python2.7/site-packages/haystack/utils/highlighting.py", line 29, in highlight
    return self.render_html(highlight_locations, start_offset, end_offset)
  File "/home/ubuntu/workspace/zds-site/zds/utils/highlighter.py", line 14, in render_html
    return super(SearchHighlighter, self).render_html()
  File "/home/ubuntu/workspace/zds-site/zdsenv/local/lib/python2.7/site-packages/haystack/utils/highlighting.py", line 117, in render_html
    for term, locations in highlight_locations.items():
AttributeError: 'NoneType' object has no attribute 'items'
```

--- a/zds/settings.py
+++ b/zds/settings.py
@@ -351,6 +351,7 @@ HAYSTACK_CONNECTIONS = {
         # 'URL': 'http://127.0.0.1:8983/solr/mysite',
     },
 }
+HAYSTACK_CUSTOM_HIGHLIGHTER = 'zds.utils.highlighter.SearchHighlighter'
 
 GEOIP_PATH = os.path.join(BASE_DIR, 'geodata')
 

--- a/zds/utils/highlighter.py
+++ b/zds/utils/highlighter.py
@@ -1,0 +1,19 @@
+# coding: utf-8
+from haystack.utils import Highlighter
+
+
+MAX_WRAP_TEXT = 50
+
+
+class SearchHighlighter(Highlighter):
+
+    def render_html(self, highlight_locations=None, start_offset=None, end_offset=None):
+
+        if start_offset is not None:
+            start_offset = max([0, start_offset - MAX_WRAP_TEXT])
+        if end_offset is not None:
+            end_offset = min([len(self.text_block), end_offset + MAX_WRAP_TEXT])
+
+        return super(SearchHighlighter, self).render_html(highlight_locations=highlight_locations,
+                                                          start_offset=start_offset,
+                                                          end_offset=end_offset)

